pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -825,3 +835,34 @@ func newCloud(sess *session.Session, l cloudprovider.Logger) (*Cloud, error) {
 	}
 	return c, nil
 }
+
+// validateMachineType checks whether the given machine type is valid and returns true if so
+func validateMachineType(machine string) bool {
+	// These are currently hardcoded as AWS does not expose them via the API
+	machinePrefixes := []string{"t2", "m3", "m4"}
```

Comment:
These should be constants (of `var` type to be precise) at the top of the file. There are also a lot more machine types for AWS that should be included. Not sure whether `nano` size support is valid? Would `nano` not be too tiny?